### PR TITLE
Remove extended option in System.Reflection.AssemblyNameTests.FullName

### PR DESCRIPTION
**Pull request by CoffeeFlux**
Fixes https://github.com/dotnet/corefx/issues/38615

### CODE REVIEW

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 1]
```diff
@@ -302,10 +302,9 @@ public void FullName(string name, string expectedName)
             AssemblyName assemblyName = new AssemblyName(name);
 
             expectedName = expectedName.ToLowerInvariant();
```
Comment (jkotas) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: Nit: Are the `ToLowerInvariant` calls needed - remove them too? I would not expect that assembly name parsing and formatting flips the case.

Comment (CoffeeFlux) [reply src/System.Reflection/tests/AssemblyNameTests.cs]: I thought that too initially, but they were added in https://github.com/dotnet/corefx/commit/617a9e944c3946ced0c2a671ce9ec08744de9a4b so I kept them around.

Comment (jkotas) [reply src/System.Reflection/tests/AssemblyNameTests.cs]: That change was compensating for .NET Framework bug. We do not run these tests on .NET Framework anymore. It should be fine to remove these.

Comment (CoffeeFlux) [reply src/System.Reflection/tests/AssemblyNameTests.cs]: Ah okay, will do.

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 2]
```diff
@@ -301,11 +301,10 @@ public void FullName(string name, string expectedName)
         {
             AssemblyName assemblyName = new AssemblyName(name);
 
-            expectedName = expectedName.ToLowerInvariant();
-            string extended = $"{expectedName}, Culture=neutral, PublicKeyToken=null".ToLowerInvariant();
-            string afn = assemblyName.FullName.ToLowerInvariant();
+            expectedName = expectedName;
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: This line is a nop.  Just delete it?

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 3]
```diff
@@ -301,11 +301,10 @@ public void FullName(string name, string expectedName)
...
+            string afn = assemblyName.FullName;
 
-            Assert.True(afn == expectedName || afn == extended, $"Expected\n{afn} == {expectedName}\nor\n{afn} == {extended}");
+            Assert.True(afn == expectedName, $"Expected\n{afn} == {expectedName}");
```
Comment (stephentoub) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: Seems like the whole test should just be:
```C#
Assert.Equal(expectedName, new AssemblyName(name).FullName);
```

File: src/System.Reflection/tests/AssemblyNameTests.cs (hunk 2 above)
Comment (CoffeeFlux) [reply src/System.Reflection/tests/AssemblyNameTests.cs]: Oh yeah, sorry... that's what I get for just blindly running a find and replace on the snippet.

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 4]
```diff
@@ -300,12 +300,7 @@ public void Flags_CurrentlyExecutingAssembly()
         public void FullName(string name, string expectedName)
         {
             AssemblyName assemblyName = new AssemblyName(name);
-
-            expectedName = expectedName.ToLowerInvariant();
-            string extended = $"{expectedName}, Culture=neutral, PublicKeyToken=null".ToLowerInvariant();
-            string afn = assemblyName.FullName.ToLowerInvariant();
-
-            Assert.True(afn == expectedName || afn == extended, $"Expected\n{afn} == {expectedName}\nor\n{afn} == {extended}");
+            Assert.Equal(expectedName, assemblyName.FullName);
```
Comment (jkotas) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: There is another `ToLowerInvariant` around line 50 that can be deleted as well.

**Comment by CoffeeFlux** [comment]
Noticed another extended option that relied on the same legacy behavior, so I got rid of that as well as the other ToLowerInvariant call.

File: src/System.Reflection/tests/AssemblyNameTests.cs [hunk 5]
```diff
@@ -407,10 +402,8 @@ public void Version(Version version, string versionString)
             assemblyName.Version = version;
 
             string expected = "MyAssemblyName, Version=" + versionString;
-            string extended = expected + ", Culture=neutral, PublicKeyToken=null";
 
-            Assert.True(assemblyName.FullName == expected || assemblyName.FullName == extended,
-                        $"Expected\n{assemblyName.FullName} == {expected}\nor\n{assemblyName.FullName} == {extended}");
+            Assert.Equal(assemblyName.FullName, expected);
```
Comment (jkotas) [inline src/System.Reflection/tests/AssemblyNameTests.cs]: This first argument for `Assert.Equal` is expected, the second argument is actual.

**Comment by CoffeeFlux** [comment]
Sorry again for all the trouble - one of those days...

**Pull request merged**